text/0715-construct-upgrade-support.md

```diff
@@ -0,0 +1,1274 @@
+to above, the main resources can be determined based on the `--target` value.
+
+There may be future enhancement that we add to the change set validation, please see
+[Appendix D: Change Set Analysis Future Enhancements](#appendix-d-change-set-analysis-future-enhancements).
+
+#### [In-Place Migration Only] CloudFormation stack refactoring Validation
+
+For in-place migrations, the construct-upgrade CLI command ensures the correctness of logical ID mappings
+defined in `refactor.json` by validating against the deployed V1 template and synthesized V2 template.
+
+```json
+[
+    {
+        "Source": {
+            "StackName": "<stack-name>",
+            "LogicalResourceId": "vpcv1A85508D7"
+        },
+        "Destination": {
+            "StackName": "<stack-name>",
+            "LogicalResourceId": "vpcA2121C38"
+        }
+    },
+    {
+        "Source": {
+            "StackName": "<stack-name>",
+            "LogicalResourceId": "vpcv1ipv6cidr87175A16"
+        },
+        "Destination": {
+            "StackName": "<stack-name>",
+            "LogicalResourceId": "vpcamazonProvided373FDA5C"
+        }
+    },
+    ......
+]
+```
+
+The validation begins by verifying that every source LogicalId in `refactor.json` corresponds to a resource
+in the deployed V1 template, while every target LogicalId maps to a resource in the synthesized V2 template.
+
+Orphaned entries — such as unmapped logical IDs or resources missing from either template — will block the upgrade.
+Resource types must also align: for example, a resource in V1 template of type `AWS::EC2::VPC` can only map to
+a resource in V2 template of a compatible type (e.g., `AWS::EC2::VPC`).
+
+This validation is specific to the context of construct upgrade framework and will be implemented as part of
+`cdk construct-upgrade --unstable=construct-upgrade` CLI command.
+
+> Note: CDK team is working on a [cdk refactor feature](https://github.com/aws/aws-cdk-rfcs/pull/705)
+> to support CloudFormation Stack refactoring with CDK CLI. Once implemented, we will onboard to use CDK
+> refactor feature instead and users do not need to manually create the `refactor.json` file and will be
+> inferred based on the construct tree and paths.
+>
+> This step is still needed to validate the inferred `refactor.json` file by CDK refactor. Depending on
+> the CDK refactor implementation, we may omit this validation if it's already performed in CDK refactor;
+> or we can move the validation logics to be a generic validation on CDK refactor side.
+
+#### [In-Place Migration Only] Path Metadata Validation
+
+In In-Place migrations, we leverage the CloudFormation stack refactoring feature, which imposes a limitation:
+no changes to resource configurations are allowed, including updates to the Metadata section of any resource.
+
+A common issue arises when users migrate a construct across modules (i.e. `@aws-cdk.aws-ec2-alpha.VpcV2` vs
```

Comment:
Like we discussed, this only depends on the logical ID changing, not the module.